sdk/search/azure-search/src/main/java/com/azure/search/documents/implementation/util/CustomPatternAnalyzerDeserializer.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation.util;
+
+import com.azure.search.documents.models.PatternAnalyzer;
+import com.azure.search.documents.models.PatternTokenizer;
+import com.azure.search.documents.models.RegexFlags;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Custom deserializer for {@link PatternAnalyzer}, which convert flatten string to the list of {@link RegexFlags}.
+ */
+public class CustomPatternAnalyzerDeserializer extends JsonDeserializer<PatternAnalyzer> {
+    private static final String DELIMITER = "\\|";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PatternAnalyzer deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode root = mapper.readTree(p);
+        JsonNode flattenFlags = root.get("flags");
+        List<RegexFlags> regexFlags = Arrays.stream(flattenFlags.asText().split(DELIMITER))
```

Comment:
What happens if `flattenFlags` is null?

Reply:
That's a miss. Will double check null cases for all.